### PR TITLE
client-go: update tikv client-go for batch_client auto re-connect idle connection when wait connection ready

### LINKED ISSUE #44471: select * from information_schema.tables get `Error 1105 (HY000): context deadline exceeded`
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. import data from `br`
2. inject chaos to shutdown `pd`, `tidb`, `tikv` for 5 minutes in order, then delete chaos (not sure if it is necessary）
3. run query ```SELECT * FROM `information_schema`.`tables` WHERE `table_schema` = ? AND `table_name` = ? LIMIT 1;```

### 2. What did you expect to see? (Required)
query run success
### 3. What did you see instead (Required)
```
<*mysql.MySQLError | 0xc000f1c000>: {
          Number: 1105,
          SQLState: [72, 89, 48, 48, 48],
          Message: "context deadline exceeded",
}
Error 1105 (HY000): context deadline exceeded
```

some related log
```
[2023/06/06 09:59:52.387 +00:00] [INFO] [conn.go:1181] ["command dispatched failed"] [conn=7442482058229717603] [connInfo="id:7442482058229717603, addr:172.30.14.77:53262 status:10, collation:utf8mb4_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="select * from `information_schema` . `tables` where `table_schema` = ? and `table_name` = ? limit ? ;"] [txn_mode=PESSIMISTIC] [timestamp=0] [err="context deadline exceeded\ngithub.com/tikv/client-go/v2/internal/client.sendBatchRequest\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.4-0.20230529120907-22ecfce6dfe8/internal/client/client_batch.go:789\ngithub.com/tikv/client-go/v2/internal/client.(*RPCClient).sendRequest\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.4-0.20230529120907-22ecfce6dfe8/internal/client/client.go:497\ngithub.com/tikv/client-go/v2/internal/client.(*RPCClient).SendRequest\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.4-0.20230529120907-22ecfce6dfe8/internal/client/client.go:540\ngithub.com/tikv/client-go/v2/internal/client.interceptedClient.SendRequest.func1\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.4-0.20230529120907-22ecfce6dfe8/internal/client/client_interceptor.go:39\ngithub.com/pingcap/tidb/util/topsql/stmtstats.(*KvExecCounter).RPCInterceptor.func1.1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/util/topsql/stmtstats/kv_exec_count.go:57\ngithub.com/tikv/client-go/v2/internal/client.interceptedClient.SendRequest\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.4-0.20230529120907-22ecfce6dfe8/internal/client/client_interceptor.go:40\ngithub.com/tikv/client-go/v2/internal/client.reqCollapse.SendRequest\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.4-0.20230529120907-22ecfce6dfe8/internal/client/client_collapse.go:74\ngithub.com/tikv/client-go/v2/internal/locate.(*RegionRequestSender).sendReqToRegion\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.4-0.20230529120907-22ecfce6dfe8/internal/locate/region_request.go:1217\ngithub.com/tikv/client-go/v2/internal/locate.(*RegionRequestSender).SendReqCtx\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.4-0.20230529120907-22ecfce6dfe8/internal/locate/region_request.go:1046\ngithub.com/tikv/client-go/v2/txnkv/txnsnapshot.(*ClientHelper).SendReqCtx\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.4-0.20230529120907-22ecfce6dfe8/txnkv/txnsnapshot/client_helper.go:146\ngithub.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleTaskOnce\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/store/copr/coprocessor.go:1041\ngithub.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleTask\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/store/copr/coprocessor.go:943\ngithub.com/pingcap/tidb/store/copr.(*copIteratorWorker).run\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/store/copr/coprocessor.go:653\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_arm64.s:1172\ngithub.com/pingcap/errors.AddStack\n\t/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20220729040631-518f63d66278/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20220729040631-518f63d66278/juju_adaptor.go:15\ngithub.com/pingcap/tidb/store/driver/error.ToTiDBErr\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/store/driver/error/error.go:169\ngithub.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleTaskOnce\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/store/copr/coprocessor.go:1042\ngithub.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleTask\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/store/copr/coprocessor.go:943\ngithub.com/pingcap/tidb/store/copr.(*copIteratorWorker).run\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/store/copr/coprocessor.go:653\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_arm64.s:1172"]
[2023/06/06 09:58:42.239 +00:00] [ERROR] [domain.go:763] ["reload schema in loop failed"] [error="context deadline exceeded"]
[2023/06/06 09:58:42.239 +00:00] [WARN] [backoff.go:158] ["tikvRPC backoffer.maxSleep 40000ms is exceeded, errors:\nsend tikv request error: context deadline exceeded, ctx: region ID: 19073, meta: id:19073 end_key:\"t\\200\\000\\000\\000\\000\\000\\000(_i\\200\\000\\000\\000\\000\\000\\000\\001\\003\\200\\000\\000\\000\\000\\000\\n\\245\\003\\200\\000\\000\\000\\000\\000\\000\\000\\003\\200\\000\\000\\000\\000\\000\\000\\005\\003\\200\\000\\000\\000\\000\\005\\264\\241\" region_epoch:<conf_ver:5 version:52 > peers:<id:19074 store_id:1 > peers:<id:19075 store_id:5 > peers:<id:19076 store_id:2 > , peer: id:19074 store_id:1 , addr: db-tikv-2.db-tikv-peer.tidb1373933076662347175.svc:20160, idx: 0, reqStoreType: TiKvOnly, runStoreType: tikv, try next peer later at 2023-06-06T09:58:12.226515675Z\nsend tikv request error: context deadline exceeded, ctx: region ID: 19073, meta: id:19073 end_key:\"t\\200\\000\\000\\000\\000\\000\\000(_i\\200\\000\\000\\000\\000\\000\\000\\001\\003\\200\\000\\000\\000\\000\\000\\n\\245\\003\\200\\000\\000\\000\\000\\000\\000\\000\\003\\200\\000\\000\\000\\000\\000\\000\\005\\003\\200\\000\\000\\000\\000\\005\\264\\241\" region_epoch:<conf_ver:5 version:52 > peers:<id:19074 store_id:1 > peers:<id:19075 store_id:5 > peers:<id:19076 store_id:2 > , peer: id:19074 store_id:1 , addr: db-tikv-2.db-tikv-peer.tidb1373933076662347175.svc:20160, idx: 0, reqStoreType: TiKvOnly, runStoreType: tikv, try next peer later at 2023-06-06T09:58:22.230749465Z\nsend tikv request error: context deadline exceeded, ctx: region ID: 19073, meta: id:19073 end_key:\"t\\200\\000\\000\\000\\000\\000\\000(_i\\200\\000\\000\\000\\000\\000\\000\\001\\003\\200\\000\\000\\000\\000\\000\\n\\245\\003\\200\\000\\000\\000\\000\\000\\000\\000\\003\\200\\000\\000\\000\\000\\000\\000\\005\\003\\200\\000\\000\\000\\000\\005\\264\\241\" region_epoch:<conf_ver:5 version:52 > peers:<id:19074 store_id:1 > peers:<id:19075 store_id:5 > peers:<id:19076 store_id:2 > , peer: id:19074 store_id:1 , addr: db-tikv-2.db-tikv-peer.tidb1373933076662347175.svc:20160, idx: 0, reqStoreType: TiKvOnly, runStoreType: tikv, try next peer later at 2023-06-06T09:58:32.235082395Z\nlongest sleep type: tikvRPC, time: 41181ms"]
```
### 4. What is your TiDB version? (Required)
```log
["Welcome to TiDB."] ["Release Version"=v6.5.3] [Edition=Enterprise] ["Git Commit Hash"=e954f278eba827068e8ec0cfbdcb5299bd0986e0] ["Git Branch"=heads/refs/tags/v6.5.3] ["UTC Build Time"="2023-06-05 10:26:58"] [GoVersion=go1.19.9] ["Race Enabled"=false] ["Check Table Before Drop"=false] ["TiKV Min Version"=6.2.0-alpha]
```

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

Issue Number: Ref #44431, close #44471

Problem Summary:

Related PR: https://github.com/tikv/client-go/pull/833

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -266,3 +266,5 @@ replace (
 	github.com/pingcap/tidb/parser => ./parser
 	go.opencensus.io => go.opencensus.io v0.23.1-0.20220331163232-052120675fac
 )
+
+replace github.com/tikv/client-go/v2 => github.com/crazycs520/client-go/v2 v2.0.0-alpha.0.20230609110852-067ec9341f7b
```
Comment (hawkingrei) [inline go.mod:270]: Please run ```make bazel_prepare```.

Comment (crazycs520) [reply go.mod:270]: I will update this after https://github.com/tikv/client-go/pull/833 merged.

**Review by zyguan — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**